File: BOJ/BOJ2477.java
package com.ssafy.day01;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.ArrayList;
import java.util.StringTokenizer;

public class BOJ2477 {
	
	static class Pair{
		int x;
		int y;
		
		public Pair() {
			
		}
		
		public Pair(int x, int y) {
			this.x = x;
			this.y = y;
		}
	}

	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
		StringTokenizer st = new StringTokenizer(br.readLine());
		
		int K, answer = 0;
		boolean flag = false;
		ArrayList<Pair> p = new ArrayList<Pair>();
		K = Integer.parseInt(st.nextToken());
		for (int i = 0; i < 6; i++) {
			st = new StringTokenizer(br.readLine());
			p.add(new Pair(Integer.parseInt(st.nextToken()), Integer.parseInt(st.nextToken())));
		}
		
		for (int i = 0; i < 3; i++) {
			if (p.get(i).x == p.get(i+2).x && p.get(i+1).x == p.get(i+3).x) {
				answer = ((p.get(i).y + p.get(i+2).y) * (p.get(i+1).y + p.get(i+3).y)) - (p.get(i+1).y * p.get(i+2).y);
				flag = true;
				break;
			}
		}
		
		if (flag) {
			bw.write(answer*K + "\n");
		}
		else {
			if (p.get(0).x == p.get(2).x) {
				answer = ((p.get(0).y + p.get(2).y) * (p.get(1).y + p.get(5).y)) - (p.get(0).y * p.get(1).y);
				bw.write(answer*K + "\n");
			}
			else if(p.get(0).x == p.get(4).x && p.get(1).x == p.get(5).x){
				answer = ((p.get(0).y + p.get(4).y) * (p.get(1).y + p.get(5).y)) - (p.get(0).y * p.get(5).y);
				bw.write(answer*K + "\n");
			}
			else {
				answer = ((p.get(0).y + p.get(4).y) * (p.get(3).y + p.get(5).y)) - (p.get(4).y * p.get(5).y);
				bw.write(answer*K + "\n");
			}
		}
		br.close();
		bw.flush();
		bw.close();
	}

}

File: BOJ/BOJ2477.cpp
//#include<iostream>
//
//using namespace std;
//
//int main() {
//	ios_base::sync_with_stdio(false);
//	cin.tie(0);
//	cout.tie(0);
//	
//	int K, answer = 0;
//	bool flag = false;
//	pair<int, int> p[6];
//	cin >> K;
//	for (int i = 0; i < 6; i++) {
//		cin >> p[i].first >> p[i].second;
//	}
//	for (int i = 0; i < 3; i++) {
//		//현재 위치와 다다음 위치와의 방향이 같고 다음 위치와 다다다음 위치의 방향이 같다면
//		if (p[i].first == p[i + 2].first && p[i+1].first == p[i+3].first) {
//			//밑의 식이 성립됨
//			answer = ((p[i].second + p[i + 2].second) * (p[i + 1].second + p[i + 3].second)) - (p[i + 1].second * p[i + 2].second);
//			flag = true;
//			break;
//		}
//	}
//
//	//위의 경우에 맞다면 답 출력
//	if (flag) {
//		cout << answer*K << "\n";
//	}
//	//위의 경우가 아니면 3가지 경우로 나뉘어짐 그에 맞는 계산식을 세움
//	else {
//		if (p[0].first == p[2].first) {
//			answer = ((p[0].second + p[2].second) * (p[1].second + p[5].second)) - (p[0].second * p[1].second);
//			cout << answer* K << "\n";
//		}
//		else if(p[0].first == p[4].first && p[1].first == p[5].first){
//			answer = ((p[0].second + p[4].second) * (p[1].second + p[5].second)) - (p[0].second * p[5].second);
//			cout << answer* K << "\n";
//		}
//		else {
//			answer = ((p[0].second + p[4].second) * (p[3].second + p[5].second)) - (p[4].second * p[5].second);
//			cout << answer * K << "\n";
//		}
//	}
//
//	return 0;
//}